alma/lib/Utils/CartDataHelper.php

```diff
@@ -0,0 +1,368 @@
+            'county_sublocality' => null,
+            'state_province' => $this->getStateProvince('billing'),
+        ];
+    }
+
+    /**
+     * Get birthday date
+     *
+     * @return string|null
+     */
+    private function getBirthday()
+    {
+        $birthday = $this->customer->birthday;
+        if ($birthday == '0000-00-00') {
+            return null;
+        }
+
+        return $birthday;
+    }
+
+    /**
+     * Get addresses of customer Prestashop
+     *
+     * @return array getCustomerAddresses
+     */
+    private function getCustomerAddresses()
+    {
+        if (version_compare(_PS_VERSION_, '1.5.4.0', '<')) {
+            $idLang = $this->context->language->id;
+        } else {
+            $idLang = $this->customer->id_lang;
+        }
+
+        return $this->customer->getAddresses($idLang);
+    }
+
+    /**
+     * Get addresses of customer for Alma
+     *
+     * @return array getAddressesData
+     */
+    private function getAddressesData()
+    {
+        $addresses = [];
+        
+        foreach ($this->getCustomerAddresses() as $address) {
+            array_push($addresses, [
+                'line1' => $address['address1'],
+                'postal_code' => $address['postcode'],
+                'city' => $address['city'],
+                'country' => Country::getIsoById((int) $address['id_country']),
+                'county_sublocality' => null,
+                'state_province' => $address['state'],
+            ]);
+        }
+
+        return $addresses;
+    }
+
+    /**
+     * get phone between info customer and addresses customer
+     *
+     * @return string
+     */
+    private function getPhone()
+    {
+        $phone = null;
+        $shippingAddress = $this->getCartAddressByType('shipping');
+
+        if ($shippingAddress->phone) {
+            $phone = $shippingAddress->phone;
+        } elseif ($shippingAddress->phone_mobile) {
+            $phone = $shippingAddress->phone_mobile;
+        }
+
+        if (is_null($phone)) {
+            foreach ($this->getCustomerAddresses() as $address) {
+                if ($address['phone']) {
+                    $phone = $address['phone'];
+                } elseif ($address['phone_mobile']) {
+                    $phone = $address['phone_mobile'];
+                }
+            }
+        }
+
+        return $phone;
+    }
+
+    /**
+     * Get customer
+     *
+     * @return Customer
+     */
+    private function getCustomer()
+    {
+        if ($this->cart->id_customer) {
+            $customer = new Customer($this->cart->id_customer);
+            if (!Validate::isLoadedObject($customer)) {
+                Logger::instance()->error(
+                    "[Alma] Error loading Customer {$this->cart->id_customer} from Cart {$this->cart->id}"
+                );
+
+                return null;
+            }
+
+            return $customer;
+        }
+
+        return $this->context->customer;
```

Comment:
why don't you return `$this->context->customer` instead of null line 244 ?
why do you try to retrieve customer with `$this->cart->id_customer` instead of getting directly the customer from the context ? (and you could retrieve the customer from the cart by id only if there is no customer into the context ?)